sklearn/impute/_iterative.py

```diff
@@ -559,7 +559,8 @@ def fit_transform(self, X, y=None):
             self._estimator = clone(self.estimator)
 
         if hasattr(self._estimator, 'random_state'):
-            self._estimator.random_state = self.random_state_
+            if self._estimator.random_state is None:
```

Comment:
Shouldn't this be
```suggestion
            if self._estimator.random_state is None and self.random_state_ is not None:
```
?

Reply:
I don't think so, because `self.random_state_` (with underscore) is returned from function `check_random_state` through `self.random_state` (no underscore), so it cannot be missing (L541).